geopandas/io/tests/test_sql.py

```diff
@@ -21,67 +24,54 @@
     from sqlalchemy import text
 except ImportError:
     # Avoid local imports for text in all sqlalchemy tests
-    # all tests using text use engine_postgis, which ensures sqlalchemy is available
+    # all tests using text use engine_psycopg_any, which ensures sqlalchemy is available
     text = str
 
 
-@pytest.fixture
-def df_nybb(nybb_filename):
-    df = read_file(nybb_filename)
-    return df
+def prepare_database_credentials() -> dict:
+    """Gather postgres connection credentials from environment variables."""
+    return {
+        "dbname": "test_geopandas",
+        "user": os.environ.get("PGUSER"),
+        "password": os.environ.get("PGPASSWORD"),
+        "host": os.environ.get("PGHOST"),
+        "port": os.environ.get("PGPORT"),
+    }
 
 
-@pytest.fixture()
-def connection_postgis():
-    """
-    Initiates a connection to a postGIS database that must already exist.
-    See create_postgis for more information.
-    """
-    psycopg2 = pytest.importorskip("psycopg2")
-    from psycopg2 import OperationalError
-
-    dbname = "test_geopandas"
-    user = os.environ.get("PGUSER")
-    password = os.environ.get("PGPASSWORD")
-    host = os.environ.get("PGHOST")
-    port = os.environ.get("PGPORT")
+def connection_psycopg_any(psycopg_package_name: str):
+    """Create a postgres connection using either psycopg2 or psycopg."""
+    psycopg = pytest.importorskip(psycopg_package_name)
+
     try:
-        con = psycopg2.connect(
-            dbname=dbname, user=user, password=password, host=host, port=port
-        )
-    except OperationalError:
+        con = psycopg.connect(**prepare_database_credentials())
+    except psycopg.OperationalError:
         pytest.skip("Cannot connect with postgresql database")
     with warnings.catch_warnings():
         warnings.filterwarnings(
             "ignore", message="pandas only supports SQLAlchemy connectable.*"
         )
         yield con
-    con.close()
+        con.close()
 
 
-@pytest.fixture()
-def engine_postgis():
+def engine_psycopg_any(psycopg_package_name: str):
```

Comment:
```suggestion
def engine_psycopg(psycopg_package_name: str):
```
I think the fact that this works for multiple versions is implied by the fact that it takes a package name as an argument.